docs/UserGuide.md

```diff
@@ -195,71 +198,92 @@ Examples of usage:
+* The `INDEX` used for each student is not fixed. It depends on the number shown on the displayed list of students.
 * You can use <a href='#finding-a-contact-find'>`find`</a> to get his/her `INDEX` and then apply `delete`.
 * This can save you time scrolling down an entire list to get his/her `INDEX`!
 
+### Undoing an action
+Undo an executed `add`, `edit` or `delete` command.
+
+Format: `undo`
+* Undoes only the last executed add, edit or delete command.
+* You can only use the undo command after executing an add, edit, or delete command.
+* Cannot be used in succession to undo previously executed commands besides the last executed one.
+
+### Redoing an action
+Reverses an executed undo command.
+
+Format: `redo`
+* Reverses only the last executed `undo` command.
+* You can only use the `redo` command after executing an `undo` command.
+* Cannot be used in succession to reverse previously executed `undo` commands besides the last executed one.
+
 ### Finding a contact
 Find a particular contact in Tracey and retrieve their specific details.
 
 Format: `find NAME`
-* Returns the student(s) with their student specific **details**.
+* Returns the student(s) whose name(s) contain the input keyword `NAME` with their **details**.
 * The **details** provided are the student's `Name`, `Matriculation Number`, `Faculty`, `Block`, `Phone`, `Address`, `Email` amd `Covid Status`.
-* A screenshot (Figure 1) is provided below to demonstrate what the find function returns to the user.
-* The search is case-insensitive. e.g `johnDoe` will match `Johndoe`
-* The search does not require full name. e.g. `jo` will match `Joe` and `John`
+* A screenshot (Figure 1) is provided below to demonstrate what the find function returns.
+* The search is case-insensitive. e.g `johnDoe` will match `Johndoe`.
+* The search does not require full name. e.g. `jo` will match `Joe` and `John`.
 
 Examples of usages:
 
-* `find John` returns `john` and `John Doe`
-* `find alex david` returns `Alex Yeoh` and`David Li`
-* `find J` returns `john` and`jay`
+* `find John` returns `john` and `John Doe`.
+* `find alex david` returns `Alex Yeoh` and`David Li`.
+* `find J` returns `john` and`jay`.
 
 Scenario:
 ![find](images/FindScreenshotForUG.png) *Figure 1: Screenshot of result of the find command*
 
 As seen in the *figure 1*, after the user keys in `find Bernice`, the application returns Bernice's details: `Name`, `Matriculation Number`, `Faculty`, `Block`, `Phone`, `Address`, `Email` amd `Covid Status`.
 
 ### Filtering contacts
-Filter student based on health statuses and/or faculties.
+Filter students based on faculties, health statuses and/or blocks.
 
-Format:`filter cs/HEALTH_STATUS f/FACULTY`
+Format:`filter f/FACULTY cs/COVID_STATUS b/BLOCK`
 
-* Returns a list of students with the given health status: `positive`, `negative`, `hrn`.
+* Returns a list of students with the given values in the respective specified fields.
```

Comment:
I think that the word field may be confusing for a newbie to the app. Maybe can specify that it corresponds to the type of pre-defined constants? As predefined constants are introduced in the early parts of the UG. :smile:

Reply:
Okay, you are right! I will change that ASAP.